src/main/java/seedu/address/logic/commands/RateCommand.java

```diff
@@ -0,0 +1,110 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_RATING;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_BOOKS;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.book.Book;
+import seedu.address.model.book.Rating;
+import seedu.address.model.book.exceptions.BookNotFoundException;
+import seedu.address.model.book.exceptions.DuplicateBookException;
+
+/**
+ * Changes the rating of an existing Book in the address book.
+ */
+public class RateCommand extends UndoableCommand {
+
+    public static final String COMMAND_WORD = "rate";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the rating of the book identified "
+            + "by the index number used in the last book listing. "
+            + "Existing rating will be overwritten by the input.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_RATING + "[RATING]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_RATING + "-1";
+
+    public static final String MESSAGE_ADD_RATING_SUCCESS = "Added rating to Book: %1$s";
+    public static final String MESSAGE_DELETE_RATING_SUCCESS = "Removed rating from Book: %1$s";
+
+    private final Index index;
+    private final Rating rating;
+
+    private Book bookToEdit;
+    private Book editedBook;
+
+    /**
+     * @param index of the book in the filtered book list to edit the rating
+     * @param rating of the book to be updated to
+     */
+    public RateCommand(Index index, Rating rating) {
+        requireNonNull(index);
+        requireNonNull(rating);
+
+        this.index = index;
+        this.rating = rating;
+    }
+
+    @Override
+    public CommandResult executeUndoableCommand() throws CommandException {
+        requireNonNull(bookToEdit);
+        requireNonNull(editedBook);
+
+        try {
+            model.updateBook(bookToEdit, editedBook);
+        } catch (DuplicateBookException dpe) {
+            throw new AssertionError("Changing target Book's rating should not result in a duplicate");
+        } catch (BookNotFoundException pnfe) {
+            throw new AssertionError("The target Book cannot be missing");
+        }
+        model.updateFilteredBookList(PREDICATE_SHOW_ALL_BOOKS);
+
+        return new CommandResult(generateSuccessMessage(editedBook));
+    }
+
+    @Override
+    protected void preprocessUndoableCommand() throws CommandException {
```

Comment:
```
if (model.getActiveListType() != ActiveListType.BOOK_SHELF) {
       throw new CommandException(MESSAGE_WRONG_ACTIVE_LIST)
}
```
Add a string MESSAGE_WRONG_ACTIVE_LIST.

Can refer to DeleteCommand.

Reply:
added successfully